README.txt


Step 1.
-------
Create an "Layout template" record in any page or sys-folder.

Fill out the form with the relative paths "/fileadmin/..." to your BE and
FE templates. You can use the "EXT:" prefix to specify templates in your
custom extension.

The default example templates are:

FE-Template
EXT:kb_nescefe/Resources/Private/Templates/Frontend/TwoColumns.html
or
EXT:kb_nescefe/Resources/Private/Templates/Bootstrap/TwoColumns.html

BE-Template
EXT:kb_nescefe/Resources/Private/Templates/Backend/TwoColumns.html



Step 2.
-------
Insert the following Page TS-Config on your root-page:

TCEFORM.tt_content.kbnescefe_layout {
  PAGE_TSCONFIG_ID = UID_OF_CREATED_FOLDER_WITH_TEMPLATES_GOES_HERE
}

of course replace the UID...GOES_HERE marker with the UID of the page in which you
created the "Layout template" records ....


Step 3.
-------
Then just open up the Web>Page module with some page and create a content element.
From the pulldown box where you can select the type (Text, Text w. Image, etc.)
select "Plugin" and then choose "Content elements container" as plugin type.

Of course you can use the "New content element" wizard which should be enabled
by default. Just select "Nested content element container" from the
"Typical page content" tab.

In the Tab "Plugin" of the content element choose a layout you wish to use from
the field "Layout template" and then save and exit the record.


Step 4.
-------
In the Web>Page module you will be able to fill the container with content
elements.


Step 5.
-------
View the results in the FE.


------------------------------------------------
See the example templates in the extensions folder "Resources/Private/Templates"
on how to create a template.

Sections are not supported currently so the "HorizontalColumns" template will
not work. Feel free to sponsor the reimplementation of this feature.


REQUIRED TS-Config:
-------------------

TCEFORM.tt_content.kbnescefe_layout {
  PAGE_TSCONFIG_ID = ###UID_OF_CREATED_FOLDER_WITH_TEMPLATES_GOES_HERE###
}



ChangeLog
2016-02-02 Bernhard Kraft  <kraftb@think-open.at>

	* [BUGFIX] Fix TYPO3 7.6 LTS compatibility

2015-01-12 Bernhard Kraft  <kraftb@think-open.at>

	* [BUGFIX] Error when no layout template gets selected

2015-01-07 Bernhard Kraft  <kraftb@think-open.at>

	* [BUGFIX] Frontend didn't render properly

2015-01-01 Bernhard Kraft  <kraftb@think-open.at>

	* Updated for TYPO3 6.2 compatibility
	* Major rewrite of many classes using extbase
	* Use fluid templating engine
	* Whole change-of-concept for template handling
	* Creation of functional unit tests for content element operations (DataHandler actions)

2012-07-13 Bernhard Kraft  <kraftb@think-open.at>

	* Fixed issue with clipboard XCLASS: http://www.toco3.com/de/toco3_blog/blog-post/2011/03/18/typo3-45-und-kb-nescefe-und-security-token.html
	* Uploaded extension to TER as 1.1.2

2010-12-27 Bernhard Kraft  <kraftb@think-open.at>

	* Fixed bug #10277: Content elements not localized
	* Added an option to override "sys_language_contentOL" for elements inside containers
	* Fixed bug #10202: Pasting records the language, colPos and order get lost

2010-11-24 Bernhard Kraft  <kraftb@think-open.at>

	* Fixed bug #11052: Incompatibility with TYPO3 4.5

2010-08-06 Bernhard Kraft  <kraftb@think-open.at>

	* Fixed a bug regarding "array_slice" which seems to occur only with some PHP version (thanks to Harald Witt for reporting this bug)
	* Created doxygen documenation for "class.tx_kbnescefe_layout.php"
	* Set "public" and "protected" keywords for methods in "class.tx_kbnescefe_layout.php"
	* Fixed the labels and confirm messages for pasting a content element into content columns or container columns

2010-07-23 Bernhard Kraft  <kraftb@think-open.at>

	* Changed a "width" style for special 4.4.x CSS
	* Uploaded Extensio to TER as 1.1.0
	* Updated documentation

2010-07-08 Bernhard Kraft  <kraftb@think-open.at>

	* Fixed styles issues with 4.4 branch
	* Fixed issues when moving content elements
	* Fixed issues when creating new content elements in branches < 4.3
	* Major rewrite of the tcemain hook class and copy/cut & paste handling
	* Updated documentation
	* Added a feature which allows to include custom CSS files for every BE-Template
	* Added some doxygen documentation

2010-06-29 Bernhard Kraft  <kraftb@think-open.at>

	* Making an Import/Export of kb_nescefe elements or pages with such elements now works properly
	* Now works with new TYPO3 Version 4.4.0

2010-06-23 Bernhard Kraft  <kraftb@think-open.at>

	* kb_nescefe containers can now get properly exported / imported along with all content elements inside them (Sponsored by bluemars.net)

2010-05-10 Bernhard Kraft  <kraftb@think-open.at>

	* Uploaded Extension to TER as 1.0.3

2010-03-24 Bernhard Kraft  <kraftb@think-open.at>

	* Fixed a bug: Double container element header output in FE
	* Fixed a bug: When a page containing content elements got copied and recursiveCopy is active, sub elements were not referenced correctly (THX to Erich Reitmeier and Igor Reitmair for reporting this bug and Igor Reitmeir for sponsoring the extension)

2010-02-15 Bernhard Kraft  <kraftb@think-open.at>

	* Removed static templates and included minimal required TypoScript by default.
	* Added TypoScript top level object “tt_content” as default renderObj

2010-02-11 Bernhard Kraft  <kraftb@think-open.at>

	* Created an update wizard which allows to update systems using kb_nescefe < 1.0.0 to the current version (See documentation)
	* Fixed a bug concerning creation of translated containers
	* The colPos value being set for elements inside containers is now configurable from within the extension manager

2010-02-09 Bernhard Kraft  <kraftb@think-open.at>

	* Raised extension state to "stable"
	* Change of major version to 1.0.0
	* The position of an element in the container is now stored in a separated field "parentPosition" instead of misusing the "colPos" field
	* Now works with TYPO3 version >= 4.3.0

2009-01-23 Bernhard Kraft  <kraftb@think-open.at>

	* Fixed a bug with content-slide (THX to Georg Ringer for reporting this)
	* Added Icon for "Container Element" in TCEforms CType drop-down
	* Added Icon in "new Content Element wizard"

2008-02-28 Bernhard Kraft  <kraftb@kraftb.at>

	* Removed a debug "print_r"

2007-10-02 Bernhard Kraft  <kraftb@kraftb.at>

	* Initial release



ext_emconf.php
<?php

########################################################################
# Extension Manager/Repository config file for ext "kb_nescefe".
#
# Auto generated 23-07-2010 10:59
#
# Manual updates:
# Only the data in the array - everything else is removed by next
# writing. "version" and "dependencies" must not be touched!
########################################################################

$EM_CONF[$_EXTKEY] = array(
	'title' => 'KB Nested Content Elements',
	'description' => 'Allows to create content element containers for other content elements. Works similar to gridelements or TV/Flux content columns only it is more easier to configure. kb_nescefe: Nested content elements since 2007!',
	'category' => 'plugin',
	'version' => '2.1.1',
	'state' => 'stable',
	'author' => 'Bernhard Kraft',
	'author_email' => 'kraftb@think-open.at',
	'author_company' => 'think-open.at GmbH',
	'constraints' => array(
		'depends' => array(
			'typo3' => '7.6.0-7.6.99'
		),
	),
);

